hadoop-hdds/container-service/src/main/java/org/apache/hadoop/ozone/container/replication/ReplicationSupervisorMetrics.java

```diff
@@ -0,0 +1,72 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ * <p>
+ * http://www.apache.org/licenses/LICENSE-2.0
+ * <p>
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.hadoop.ozone.container.replication;
+
+import org.apache.hadoop.hdds.annotation.InterfaceAudience;
+import org.apache.hadoop.metrics2.MetricsCollector;
+import org.apache.hadoop.metrics2.MetricsSource;
+import org.apache.hadoop.metrics2.MetricsSystem;
+import org.apache.hadoop.metrics2.annotation.Metrics;
+import org.apache.hadoop.metrics2.lib.DefaultMetricsSystem;
+import org.apache.hadoop.metrics2.lib.Interns;
+import org.apache.hadoop.ozone.OzoneConsts;
+
+/**
+ * Metrics source to report number of replication tasks.
+ */
+@InterfaceAudience.Private
+@Metrics(about = "Container Replication Supervisor Metrics",
+    context = OzoneConsts.OZONE)
+public class ReplicationSupervisorMetrics implements MetricsSource {
+
+  private static final String SOURCE =
+      ReplicationSupervisorMetrics.class.getSimpleName();
+  private final ReplicationSupervisor supervisor;
+
+  public ReplicationSupervisorMetrics(ReplicationSupervisor
+      replicationSupervisor) {
+    this.supervisor = replicationSupervisor;
+  }
+
+  public static ReplicationSupervisorMetrics create(ReplicationSupervisor
+      supervisor) {
+    MetricsSystem ms = DefaultMetricsSystem.instance();
+    return ms.register(SOURCE, "Container Replication Supervisor Metrics",
```

Comment:
we create and register one MetricsSystem per DN, which is not a problem for typical deployments.
But i imagine it can potentially be problematic for unit tests because there are multiple DN objects in unit test heap and only one of them will successfully register.

Reply:
Other than that the code looks good to me.

Reply:
Yea,  that will be a "metrics already exists" warning in unit test, not only for this metrics, but also for other DN metries. 
What's a general way to handle this?  Maybe we can append DN's  hostname+port  to each metrics source name?  
I'm not sure how HDFS handle this.

Reply:
HDFS DataNodeMetrics.java
dnName is host name + port. So yeah, it would be a good idea to try to make them unique.
public static DataNodeMetrics create(Configuration conf, String dnName) {
    String sessionId = conf.get(DFSConfigKeys.DFS_METRICS_SESSION_ID_KEY);
    MetricsSystem ms = DefaultMetricsSystem.instance();
    JvmMetrics jm = JvmMetrics.create("DataNode", sessionId, ms);
    String name = "DataNodeActivity-"+ (dnName.isEmpty()
        ? "UndefinedDataNodeName"+ ThreadLocalRandom.current().nextInt()
            : dnName.replace(':', '-'));

    // Percentile measurement is off by default, by watching no intervals
    int[] intervals = 
        conf.getInts(DFSConfigKeys.DFS_METRICS_PERCENTILES_INTERVALS_KEY);
    
    return ms.register(name, null, new DataNodeMetrics(name, sessionId,
        intervals, jm));
  }

Reply:
https://github.com/apache/hadoop/blob/95454d821c6e4c08d49df195d7c1e0ea1aacafb9/hadoop-hdfs-project/hadoop-hdfs/src/main/java/org/apache/hadoop/hdfs/server/datanode/metrics/DataNodeMetrics.java#L259

Reply:
I see. Can we do it in a followup JIRA?  since other DN metrics also need be improved.